rust/lance-encoding/src/decoder.rs

```diff
@@ -741,6 +745,15 @@ impl CoreFieldDecoderStrategy {
                         as Box<dyn StructuralFieldScheduler>,
                 )
             }
+            DataType::Binary | DataType::Utf8 => {
+                let column_info = column_infos.expect_next()?;
+                let scheduler = Box::new(StructuralPrimitiveFieldScheduler::try_new(
+                    column_info.as_ref(),
+                    self.decompressor_strategy.as_ref(),
+                )?);
+                column_infos.next_top_level();
+                Ok(scheduler)
+            }
```

Comment:
Instead of doing this can we add `Binary` and `Utf8` to `Self::is_primitive`?

Reply:
it's a good idea, I am a bit worried doing this may cause some issues in lance version priori to 2.1, I will verify this.

Reply:
fixed.